ethabi/src/event_param.rs

```diff
@@ -1,19 +1,136 @@
 //! Event param specification.
 
-use {ParamType};
+use serde::de::{Error, MapAccess, SeqAccess, Visitor};
+use serde::{Deserialize, Deserializer};
+use std::{fmt, io};
+use ParamType;
+use {errors, Constructor, ErrorKind, Event, Function};
 
 /// Event param specification.
-#[derive(Debug, Clone, PartialEq, Deserialize)]
+#[derive(Debug, Clone, PartialEq)]
 pub struct EventParam {
 	/// Param name.
 	pub name: String,
 	/// Param type.
-	#[serde(rename="type")]
 	pub kind: ParamType,
 	/// Indexed flag. If true, param is used to build block bloom.
 	pub indexed: bool,
 }
 
+pub struct TupleParams {
+    params: Vec<EventParam>,
```

Comment:
Hm hm hm, `EventParam` does not strike me as the right thing to use here. A dedicated `TupleParam` would make more sense IMHO.